### ISSUE TITLE
Bug: Published .d.ts.map files have sources that points to unpublished files

### ISSUE THREAD

**Issue by bluwy** [typescript-eslint/typescript-eslint#11613]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

types

### Repro Code

```TypeScript
NOTE: This affects a few `@typescript-eslint/` packages. I'll take one package example:

1. Visit https://www.npmjs.com/package/@typescript-eslint/types?activeTab=code
2. Check `dist/ts-estree.d.ts.map`
3. Notice the `"sources"` points to `"../src/ts-estree.ts"` which isn't published
```

### ESLint Config

```javascript
irrelevant
```

### tsconfig

```jsonc
irrelevant
```

### Expected Result

The sourcemap `"sources"` should point to an existing file, or the sourcemap should be removed entirely.

### Actual Result

The sourcemap `"sources"` points to unpublished files, meaning that the IDE can't use it to navigate to the original types or declaration if I want to "go to source definition". Any stacktraces using the sourcemap would also point to a non-existing file, making it not possible to visit the file for debugging.

### Additional Info

I find that this affects a few packages including:

- `@typescript-eslint/types`
- `@typescript-eslint/eslint-plugin`
- `@typescript-eslint/parser`
- `@typescript-eslint/typescript-estree`
- `@typescript-eslint/type-utils`
- `@typescript-eslint/utils`

It might be easier to remove the `.d.ts.map` files entirely as the `.js` files don't have sourcemaps either.

### Versions

I'm copying my local versions here but I'm pretty sure it's relevant in the current latest releases too

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `8.43.0` |
| `@typescript-eslint/parser`            | `8.43.0` |
| `@typescript-eslint/typescript-estree` | `8.43.0` |
| `@typescript-eslint/type-utils`        | `8.43.0` |
| `@typescript-eslint/utils`             | `8.43.0` |

**Comment by JoshuaKGoldberg** [comment]
Related: #9352

I'm in favor of removing these too, for the same reasons as #9352.
